soc/Kconfig

```diff
@@ -36,6 +36,10 @@ config SOC_COMPATIBLE_NRF52X
 config SOC_COMPATIBLE_NRF52833
 	bool
 
+# Helper symbol for identification of Espressif ESP32 family of SOCs.
+config SOC_FAMILY_ESP32
```

Comment:
SoC families are usually defined in the vendor SoC Kconfig, like this:
https://github.com/zephyrproject-rtos/zephyr/blob/51a7de2beb13249804200164610358158088588e/soc/xtensa/intel_adsp/Kconfig#L6
https://github.com/zephyrproject-rtos/zephyr/blob/51a7de2beb13249804200164610358158088588e/soc/arm/xilinx_zynq7000/Kconfig#L6
...

So why is this placed here ?

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/51a7de2beb13249804200164610358158088588e/soc/arm/nordic_nrf/Kconfig#L6

Reply:
In this case the identifier aims to provide common symbol which is related to the XTENSA and RISCV architectures, as ESP32 SOCs can use both of these arch.

Reply:
@tejlmand, do you have any other suggestion considering that we want to have the same `FAMILY_ESP32` for multiple architectures?